libraries/AP_NavEKF/AP_NavEKF_Source.cpp

```diff
@@ -0,0 +1,431 @@
+/*
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_Math/AP_Math.h>
+#include <AP_DAL/AP_DAL.h>
+#include "AP_NavEKF_Source.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_NavEKF_Source::var_info[] = {
+
+    // @Param: POSXY
+    // @DisplayName: Position Horizontal Source (Primary)
+    // @Description: Position Horizontal Source (Primary)
+    // @Values: 0:None, 3:GPS, 4:Beacon, 6:ExternalNav
+    // @User: Advanced
+    AP_GROUPINFO("POSXY", 1, AP_NavEKF_Source, _source[0].posxy, (int8_t)AP_NavEKF_Source::SourceXY::GPS),
+
+    // @Param: VELXY
+    // @DisplayName: Velocity Horizontal Source
+    // @Description: Velocity Horizontal Source
+    // @Values: 0:None, 3:GPS, 4:Beacon, 5:OpticalFlow, 6:ExternalNav, 7:WheelEncoder
+    // @User: Advanced
+    AP_GROUPINFO("VELXY", 2, AP_NavEKF_Source, _source[0].velxy, (int8_t)AP_NavEKF_Source::SourceXY::GPS),
+
+    // @Param: POSZ
+    // @DisplayName: Position Vertical Source
+    // @Description: Position Vertical Source
+    // @Values: 0:None, 1:Baro, 2:RangeFinder, 3:GPS, 4:Beacon, 6:ExternalNav
+    // @User: Advanced
+    AP_GROUPINFO("POSZ", 3, AP_NavEKF_Source, _source[0].posz, (int8_t)AP_NavEKF_Source::SourceZ::BARO),
+
+    // @Param: VELZ
+    // @DisplayName: Velocity Vertical Source
+    // @Description: Velocity Vertical Source
+    // @Values: 0:None, 3:GPS, 4:Beacon, 6:ExternalNav
+    // @User: Advanced
+    AP_GROUPINFO("VELZ", 4, AP_NavEKF_Source, _source[0].velz, (int8_t)AP_NavEKF_Source::SourceZ::GPS),
+
+    // @Param: YAW
+    // @DisplayName: Yaw Source
+    // @Description: Yaw Source
+    // @Values: 0:None, 1:Compass, 2:External, 3:External with Compass Fallback
+    // @User: Advanced
+    AP_GROUPINFO("YAW", 5, AP_NavEKF_Source, _source[0].yaw, (int8_t)AP_NavEKF_Source::SourceYaw::COMPASS),
+
+#if AP_NAKEKF_SOURCE_COUNT >= 2
+    // @Param: POSXY2
```

Comment:
I vote for 2_POSXY if we have sufficient characters to spare  when the prefix is included

Reply:
I've checked the count and we have enough characters, so propose we use the following prefix convention EK3__SRCn_xxxx where n is the source index. This will ensure consistent ordering in lists.

Reply:
Recommended changes here https://github.com/priseborough/ardupilot/commits/ekf-pos-source2

Reply:
Sure, I'm totally happy to move the 2 to the other side.

Reply:
the params have been renamed so I'll mark this as resolved.